### PR TITLE
lua: fix behavior when split empty string

### PR DESCRIPTION
**Pull request by notomo**
this pr will fix the behaivior vim.split.
ref: https://github.com/neovim/neovim/pull/12420#issuecomment-638044627

## before

```lua
vim.split("", "\n") == {}
```

## after 

```lua
vim.split("", "\n") == { '' }
```

### CODE REVIEW

**Comment by tjdevries** [comment]
Do you think you could add a test as well for `apply_text_edits`?

Adding a test with an empty edit in `test/functional/plugin/lsp_spec.lua`? That way we can't mess it up next time ;)

File: test/functional/plugin/lsp_spec.lua [hunk 1]
```diff
@@ -800,13 +800,14 @@ describe('LSP', function()
         make_edit(0, 0, 0, 0, {"123"});
         make_edit(1, 0, 1, 1, {"2"});
         make_edit(2, 0, 2, 2, {"3"});
+        make_edit(3, 2, 3, 4, {""});
```
Comment (notomo) [inline test/functional/plugin/lsp_spec.lua:803]: 📝 I added a test for the following edit.

```
{
  newText = "",
  range = {
    end = {
      character = 4,
      line = 3
    },
    start = {
      character = 2,
      line = 3
    }
  }
}
```

**Review by mfussenegger — APPROVED** [review]

**Comment by tjdevries** [comment]
Awesome, thanks!

Thanks for the review @mfussenegger
